Generate reflection-free serializers

tsegismont opened the pull request:
See https://quarkus.io/blog/quarkus-metaprogramming/

tsegismont commented:
Hi @mariofusco

@franz1981 suggested to enable reflection-free serializers on this benchmark so I gave it a try but it didn't work.

With some debugging info, I found that `io.quarkus.resteasy.reactive.jackson.runtime.serialisers.GeneratedSerializersRegister.MappingModuleHolder#createMappingModule` is invoked before `io.quarkus.resteasy.reactive.jackson.runtime.ResteasyReactiveServerJacksonRecorder#recordGeneratedSerializer`.

Consequently, the Jackson module is registered without any of the generated serializers.

Is this is a known issue? Otherwise, I can file one.

mariofusco commented:
Hi @tsegismont, no this is not a known issue, please create a new one. Also please clarify how I can reproduce this issue (I'm really curious to figure out why we didn't experience this issue before).

tsegismont commented:
@mariofusco thanks for the quick reply.

Here's the bug report https://github.com/quarkusio/quarkus/issues/47571

tsegismont commented:
Updated the PR: bumped to 3.22.3 and verified that reflection-free serializers get picked up.

Thanks @mariofusco !

tsegismont commented:
@franz1981, @Sanne, @geoand PTAL

geoand commented on frameworks/Java/quarkus/resteasy-reactive-hibernate-reactive/src/main/resources/application.properties:
```diff
@@ -28,3 +28,6 @@ quarkus.vertx.prefer-native-transport=true
 quarkus.arc.context-propagation.enabled=false
+
+quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true
+quarkus.package.jar.decompiler.enabled=true
```
This last option isn't really needed

tsegismont replied:
I know, but it's convenient. I can remove it if it's a blocker

geoand replied:
Let's just comment it out as it slows down the build

franz1981 commented:
Send the PR to the actual right repo if it proves to improve performance (it should!)

Funny enough, the vertx and reactive routes variant cannot benefit of this afaik

kevmoo committed:
Dart3: update to latest SDK, general cleanup (#9918)

Also added in missing new-line chars at the end of a couple of files

inemtsev committed:
Upgrade versions 05 2025 (#9914)

* Update logback.xml

* Update logback.xml

* Update logback.xml

* Update logback.xml

* Update logback.xml

* bumped versions all around

* updated gradle version, made java settings idential between ktor tests for consistency

* removed string cleanup setting, seems to slow down rps

---------

Co-authored-by: Ilya Nemtsev <ilyanemtsev@Ilyas-MacBook-Pro.local>

eastspire committed:
Upgrade hyperlane (#9912)

* feat: update & fortune

* feat: randomNumber

* feat: max row 500

* feat: update

* feat: Fortune

* feat: fortunes

* feat: update

* feat: update

* feat: update

* feat: update

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: cache

* feat: cache

* feat: config

* feat: config

* feat: config

* feat: v4.36.1

* docs: readme

* feat: remove dyn

* docs: readme

* feat: lock

* feat: lock

* feat: lock

* feat: lock

* feat: db pool

* feat: db pool

* feat: lock

* feat: lock

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: rand

* feat: rand

* feat: rand

* feat: rand

* feat: rand

* feat: port

* feat: port

* feat: port

* feat: db

* feat: db

* feat: db

* feat: lock

* feat: lock

* feat: v4.41.0

* feat: v4.42.0

* Merge remote-tracking branch 'upstream/master'

* Merge remote-tracking branch 'upstream/master'

* feat: inline

* feat: dockerfile

* feat: v4.52.1

* feat: remove key

* remove: log

* remove: log

* feat: async

* remove: empty loop

* feat: utf8

* change: pool_size

* remove: utf8

* feat: log

* feat: log

* feat: v3.14.1

* feat: 4.56.3

* feat: 4.56.4

* feat: 4.56.5

* feat:  rename

* Merge branch 'master' of github.com:TechEmpower/FrameworkBenchmarks

* feat: speed

* feat: speed

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* feat: runtime

* feat: runtime

* feat: runtime

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: hyperlane

* feat: update

* docs: readme

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: lock

* feat: use super

* feat: update lock

* feat: update

* feat: update lock

* feat: update lock

* feat: update lock

* feat: lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

---------

Co-authored-by: 尤雨东 <83822098+ltpp-universe@users.noreply.github.com>

bymoye committed:
[Python/Blacksheep] Adjust the unit startup mode (#9886)

* Adjust the unit startup mode

* fix

* try fix

* try fix

* try fix

* try fix

* try fix

* rollback

* adjust

* fix

* try change

* try change

* fix

* Tuning

* fix

* try fix

* try fix

* add pypy

* try fix pypy

* fix

* fix: app.py -> app-socketify.py

* fix: _is_travis

* fix error

* try fix

* try fix

* try fix

* try fix

* try optimization

* try fix

* rollback

* update socketify and fix query

eastspire committed:
Upgrade hyperlane (#9920)

* feat: update & fortune

* feat: randomNumber

* feat: max row 500

* feat: update

* feat: Fortune

* feat: fortunes

* feat: update

* feat: update

* feat: update

* feat: update

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: fortunes

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: cache

* feat: cache

* feat: config

* feat: config

* feat: config

* feat: v4.36.1

* docs: readme

* feat: remove dyn

* docs: readme

* feat: lock

* feat: lock

* feat: lock

* feat: lock

* feat: db pool

* feat: db pool

* feat: lock

* feat: lock

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: db

* feat: rand

* feat: rand

* feat: rand

* feat: rand

* feat: rand

* feat: port

* feat: port

* feat: port

* feat: db

* feat: db

* feat: db

* feat: lock

* feat: lock

* feat: v4.41.0

* feat: v4.42.0

* Merge remote-tracking branch 'upstream/master'

* Merge remote-tracking branch 'upstream/master'

* feat: inline

* feat: dockerfile

* feat: v4.52.1

* feat: remove key

* remove: log

* remove: log

* feat: async

* remove: empty loop

* feat: utf8

* change: pool_size

* remove: utf8

* feat: log

* feat: log

* feat: v3.14.1

* feat: 4.56.3

* feat: 4.56.4

* feat: 4.56.5

* feat:  rename

* Merge branch 'master' of github.com:TechEmpower/FrameworkBenchmarks

* feat: speed

* feat: speed

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* feat: runtime

* feat: runtime

* feat: runtime

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* lock: toml

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: hyperlane

* feat: update

* docs: readme

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: update

* feat: lock

* feat: use super

* feat: update lock

* feat: update

* feat: update lock

* feat: update lock

* feat: update lock

* feat: lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* feat: update lock

* Merge remote-tracking branch 'upstream/master'

* feat: update

* feat: utf8

* feat: utf8

---------

Co-authored-by: 尤雨东 <83822098+ltpp-universe@users.noreply.github.com>

SentryMan committed:
[avaje-jex] tweak hikari settings (#9919)

litongjava committed:
update to tio-boot v1.9.7 (#9921)

cosmastech committed:
(Laravel) - force updating to a new number (#9915)

* Update Controller.php

* ensure random int is valid

dependabot[bot] committed:
Bump rack from 3.1.11 to 3.1.16 in /frameworks/Ruby/rack-sequel

Bumps [rack](https://github.com/rack/rack) from 3.1.11 to 3.1.16.
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/main/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/v3.1.11...v3.1.16)

---
updated-dependencies:
- dependency-name: rack
  dependency-version: 3.1.16
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #9929 from TechEmpower/dependabot/bundler/frameworks/Ruby/rack-sequel/rack-3.1.16

Bump rack from 3.1.11 to 3.1.16 in /frameworks/Ruby/rack-sequel

Sanne committed:
Update to Quarkus 3.23.2

Sanne committed:
Update readme

tsegismont committed:
Generate reflection-free serializers

See https://quarkus.io/blog/quarkus-metaprogramming/

Both of the implementations with Hibernate can benefit from it (ORM and Reactive).

Sanne commented:
I see you closed it... what's the plan? Were you able to verify the performance benefits?

tsegismont commented:
@Sanne I opened this PR on my own repo to start the conversation with Mario about a possible bug https://github.com/tsegismont/FrameworkBenchmarks/pull/1#issuecomment-2835146833

I closed it because there's no reason to merge it. It's not related to the performance evaluation.

Sanne commented:
great, thanks!
